T2_2/DiagonalMatrixExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Homework_2_2;
+public static class DiagonalMatrixExtensions
+{
+	public static DiagonalMatrix Add(this DiagonalMatrix matrix1, DiagonalMatrix matrix2)
+	{
+		int maxSize = Math.Max(matrix1.Size, matrix2.Size);
+
+		int[] diagonal1 = new int[maxSize];
+		int[] diagonal2 = new int[maxSize];
```

Comment:
Try not to allocate memory for two new arrays just to set 0 to extra elements, this is not efficient. Use only one for inside Add() method.